cmd/ipfs/main.go

```diff
@@ -473,6 +481,9 @@ func (i *cmdInvocation) setupInterruptHandler() {
 	sig := allInterruptSignals()
 
 	go func() {
+		// wait till the context is ready to be closed
+		<-ctx.ContextIsReadyToBeClosed
```

Comment:
Sounds good. 

> Is it okay to `os.Exit()` during daemon startup?

Yep, AFAIR, there's no writing anything, so should not get into a bad state. (this can get cleaner once we change how the daemon starts up.